aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -1227,6 +1279,22 @@ public <T extends DynamoDbClientBuilder> void applyDynamoDbEndpointConfiguration
     configureEndpoint(builder, dynamoDbEndpoint);
   }
 
+  /**
+   * Override the AWS client region for all AWS clients used by {@link
```

Comment:
I don't think we need to mention it's override for `DefaultAwsClientFactory`, because we will probably apply this also to the other client factories in the Iceberg codebase